Simplecov test

niewicz opened the pull request:
@gotar

niewicz committed:
test coverage added -> simplecov

niewicz committed:
.rspec added

niewicz committed:
coverage test added (simplecov), .gitignore, commented display dunctions in name_spec.rb

niewicz committed:
display changed to to_s, tests added

display functions now return string

niewicz committed:
indents fixed

wafcio commented on lib/basket.rb:
```diff
@@ -34,14 +34,16 @@ def remove(product_id, quantity)
-    puts "With VAT:\t" + "%.2f" % sum_with_vat
+  def to_s
+    sum, sum_with_vat, s_tmp = compute_bill()
+    s = ">>>BASKET\n"
```
It is better to use variable name longer than one character.

gotar commented on Gemfile:
```diff
@@ -1,5 +1,6 @@
 group :test do
+	gem "simplecov"
 	gem "rspec"
-end
```
wrong indentation

gotar commented on lib/basket.rb:
```diff
@@ -34,14 +34,16 @@ def remove(product_id, quantity)
+    s += "--------------------------------------\n"
+    s += "Sum:\t\t" + "%.2f" % sum + "\n"
+    s += "With VAT:\t" + "%.2f" % sum_with_vat + "\n"
+    return s
```
return word is not needed

gotar commented on lib/warehouse.rb:
```diff
@@ -74,4 +76,3 @@ def find_in(product_id)
   end
   
```
drop empty line

gotar commented on lib/warehouse.rb:
```diff
@@ -48,10 +48,12 @@ def remove_all
-      puts "ID:\t#{item[:item]}\tQUANTITY:\t#{item[:count]}"
+      s += "ID:\t#{item[:item]}\tQUANTITY:\t#{item[:count]}\n"
     end
+    return s
```
return word not needed

gotar commented on lib/warehouse.rb:
```diff
@@ -48,10 +48,12 @@ def remove_all
+    s = ""
     @products.each do |item|
-      puts "ID:\t#{item[:item]}\tQUANTITY:\t#{item[:count]}"
+      s += "ID:\t#{item[:item]}\tQUANTITY:\t#{item[:count]}\n"
```
your attrs should means something, `s` means nothing

gotar commented on spec/spec_helper.rb:
```diff
@@ -1,3 +1,6 @@
+require 'simplecov'
+SimpleCov.start
```
add empty line above

gotar commented on spec/unit/basket_spec.rb:
```diff
@@ -11,7 +11,6 @@
     @p2 = Product.new("Ball", 8, 0.23)
     @p3 = Product.new("Table", 102.6, 0.23)
-    p "basket", @p1.id, @p2.id, @p3.id
     @wh = Warehouse.new([@p1, @p2, @p3])
```
change vars to something more descriptive

gotar commented on spec/unit/basket_spec.rb:
```diff
@@ -198,18 +201,3 @@
   end
 
```
drop empty line

niewicz committed:
improvments from comments added, let added

in basket_spec.rb and warehouse_spec.rb let basket_init and warehouse_init
using let introduced
